### DISCUSSION TITLE
how to use assets/  files in mobile?

### DISCUSSION THREAD

**Discussion by jian308** [hajimehoshi/ebiten#3085] [Q&A]
`github.com/ebitengine/gomobile/asset` not working

**Comment by hajimehoshi** [comment]
You can use [`embed`](https://pkg.go.dev/embed) to embed data to the Go binary. This is the most portable, but might affect compilation time.

Another solution is [`AssetManager`](https://developer.android.com/reference/android/content/res/AssetManager) in the Java world, and pass the data to the Go world by [binding](https://pkg.go.dev/golang.org/x/mobile/cmd/gobind), which is available even with ebitenmobile.

**Comment by jian308** [comment]
Using embed results in an aar size several times larger, and it is hoped that direct use will be supported later.

**Reply by hajimehoshi** [reply]
Sure, gomobile/assets should be available. Let me take a look later.

**Reply by hajimehoshi** [reply]
Could you explain more details about your issue? What does not work?

FYI: https://github.com/golang/go/issues/37177 might be related.
